src/components/Main.js

```diff
@@ -1,30 +1,56 @@
 // @flow
 import React from 'react';
+import { graphql } from 'react-apollo';
+import { compose, branch, renderComponent } from 'recompose';
+import gql from 'graphql-tag';
 
 import List, { type SubmissionInfo } from './List';
 import { Box } from './Layout';
 
-const fakeData: Array<SubmissionInfo> = [
-  {
-    title: 'The Criminal Underworld Is Dropping Bitcoin for Another Currency',
-    url:
-      'https://www.bloomberg.com/news/articles/2018-01-02/criminal-underworld-is-dropping-bitcoin-for-another-currency',
-    submitter: 'fragosti',
-    comments: [],
-  },
-  {
-    title: 'Bitcoin Is The MySpace Of The Cryptocurrency World (Part 1)',
-    url:
-      'https://seekingalpha.com/article/4135035-bitcoin-myspace-cryptocurrency-world-part-1',
-    submitter: 'rgmvisser',
-    comments: [],
-  },
-];
-
-const Main = () => (
+type Props = {|
+  submissions: Array<SubmissionInfo>,
+|};
+
+// Stateless component for loading state (to avoid if-statement within Main component)
+const MainLoading = () => <Box mx={2}> ...Loading </Box>;
+
+const Main = ({ submissions }: Props) => (
   <Box mx={2}>
-    <List dataSource={fakeData} />
+    <List dataSource={submissions} />
   </Box>
 );
 
-export default Main;
+// The AllPosts graphql query
+const AllPosts = gql`
+  query AllPosts {
+    allPosts {
+      id
+      title
+      url
+      author {
+        username
+      }
+      comments {
+        id
+      }
+    }
+  }
+`;
+
+// graphql(Query) returns a Higher Order Component that injects the result of Query into the Component
```

Comment:
@fragosti cool. so, for my learning purposes, how specifically would one do this with**out** the usage of a higher order component?

Reply:
https://gist.github.com/fragosti/1eefef6e6200f4dc3332a2ee0caac092

Reply:
Oh. That isn't exactly what you were asking but hopefully it's helpful lol. 

Here is the answer to your question. 
https://www.apollographql.com/docs/react/basics/setup.html#basic-operations